tests/upgrades/test_classparameter.py

```diff
@@ -52,29 +54,40 @@ class TestScenarioPositivePuppetParameterAndDatatypeIntact:
     """
 
     @pytest.fixture(scope="class")
-    def _setup_scenario(self, target_sat, save_test_data):
+    def _setup_scenario(self, pre_upgrade_puppet_enabled_sat):
         """Import some parametrized puppet classes. This is required to make
         sure that we have smart class variable available.
         Read all available smart class parameters for imported puppet class to
         be able to work with unique entity for each specific test.
         """
-        self.org = entities.Organization().create()
+        self.org = pre_upgrade_puppet_enabled_sat.api.Organization().create()
         repo = 'api_test_classparameters'
-        env_name = target_sat.create_custom_environment(repo=repo)
-        self.puppet_class = entities.PuppetClass().search(
+        env_name = pre_upgrade_puppet_enabled_sat.create_custom_environment(repo=repo)
+        self.puppet_class = pre_upgrade_puppet_enabled_sat.api.PuppetClass().search(
             query={'search': f'name = "{repo}" and environment = "{env_name}"'}
         )[0]
-        self.sc_params_list = entities.SmartClassParameters().search(
+        self.sc_params_list = pre_upgrade_puppet_enabled_sat.api.SmartClassParameters().search(
             query={'search': f'puppetclass="{self.puppet_class.name}"', 'per_page': 1000}
         )
-        save_test_data({'puppet_class': self.puppet_class.name})
+        return {
+            'puppet_class': self.puppet_class.name,
+            'satellite': pre_upgrade_puppet_enabled_sat.hostname,
+        }
 
     @pytest.fixture(scope="class")
-    def _clean_scenario(self, request, class_target_sat, pre_upgrade_data):
+    def _clean_scenario(self, request, class_pre_upgrade_data):
         @request.addfinalizer
         def _cleanup():
-            puppet_class = pre_upgrade_data.get('puppet_class')
-            class_target_sat.delete_puppet_class(puppet_class)
+            sat_hostname = getattr(
+                class_pre_upgrade_data, next(iter(class_pre_upgrade_data))
+            ).satellite
+            puppet_class = getattr(
+                class_pre_upgrade_data, next(iter(class_pre_upgrade_data))
+            ).puppet_class
+            Satellite(sat_hostname).delete_puppet_class(puppet_class)
```

Comment:
Since we are anyways tearing down the satellite, we need to delete the puppet class.